tlslite/utils/rsakey.py

```diff
@@ -87,6 +91,164 @@ def hashAndVerify(self, sigBytes, bytes):
+        return T[:maskLen]
+
+    def EMSA_PSS_encode(self, M, emBits, hAlg, sLen=0):
+        """Encode the passed in message
+
+        This encodes the message using selected hash algorithm
+
+        @type M: bytearray
+        @param M: Message to be encoded
+
+        @type emBits: int
+        @param emBits: maximal length of returned EM
+
+        @type hAlg: str
+        @param hAlg: hash algorithm to be used
+
+        @type sLen: int
+        @param sLen: length of salt"""
+        hashLen = getattr(hashlib, hAlg)().digest_size
+        mHash = secureHash(M, hAlg)
+        emLen = Poly1305.divceil(emBits, 8)
+        if emLen < hashLen + sLen + 2:
+            raise EncodingError("Encoding error")
+        salt = getRandomBytes(sLen)
+        M2 = bytearray(8) + mHash + salt
+        H = secureHash(M2, hAlg)
+        PS = bytearray(emLen - sLen - hashLen - 2)
+        DB = PS + bytearray(b'\x01') + salt
+        dbMask = self.MGF1(H, emLen - hashLen - 1, hAlg)
+        maskedDB = bytearray(i ^ j for i, j in zip(DB, dbMask))
+        mLen = emLen*8 - emBits
+        mask = (1 << 8 - mLen) - 1
+        maskedDB[0] &= mask
+        EM = maskedDB + H + bytearray(b'\xbc')
+        return EM
+
+    def RSASSA_PSS_sign(self, M, hAlg, sLen=0):
+        """"Sign the passed in message
+
+        This signs the message using selected hash algorithm
+
+        @type M: bytearray
+        @param M: Message to be signed
+
+        @type hAlg: str
+        @param hAlg: hash algorithm to be used
+
+        @type sLen: int
+        @param sLen: length of salt"""
+        EM = self.EMSA_PSS_encode(M, numBits(self.n) - 1, hAlg, sLen)
+        m = bytesToNumber(EM)
+        assert m < self.n
+        s = self._rawPrivateKeyOp(m)
+        S = numberToByteArray(s, numBytes(self.n))
+        return S
+
+    def EMSA_PSS_verify(self, M, EM, emBits, hAlg, sLen=0):
+        """Verify signature in passed in encoded message
+
+        This verifies the signature in encoded message
+
+        @type M: bytearray
+        @param M: Original not signed message
+
+        @type EM: bytearray
+        @param EM: Encoded message
+
+        @type emBits: int
+        @param emBits: Length of the encoded message in bits
+
+        @type hAlg: str
+        @param hAlg: hash algorithm to be used
+
+        @type sLen: int
+        @param sLen: Length of salt
+        """
+        hashLen = getattr(hashlib, hAlg)().digest_size
+        mHash = secureHash(M, hAlg)
+        emLen = Poly1305.divceil(emBits, 8)
+        if emLen < hashLen + sLen + 2:
+            raise InvalidSignature("Invalid signature")
+        if EM[-1] != 0xbc:
+            raise InvalidSignature("Invalid signature")
+        maskedDB = EM[0:emLen - hashLen - 1]
+        H = EM[emLen - hashLen - 1:emLen - hashLen - 1 + hashLen]
+        for x in range(0, 8*emLen - emBits):
+            if maskedDB[0] & Poly1305.divceil(128, (x+1)) != 0:
+                raise InvalidSignature("Invalid signature")
```

Comment:
couldn't we prepare a mask with enough bits set and then check if `maskedDB[0] & mask != 0`?